src/analysis/retail/monk/mistweaver/normalizers/CastLinkNormalizer.ts

```diff
@@ -266,4 +362,48 @@ export function isFromDancingMists(event: ApplyBuffEvent | RefreshBuffEvent): bo
   return HasRelatedEvent(event, FROM_DANCING_MISTS) && !HasRelatedEvent(event, FROM_MISTS_OF_LIFE);
 }
 
+export function isFromEnvelopingMist(event: HealEvent) {
+  return HasRelatedEvent(event, ENVELOPING_MIST_GOM);
+}
+
+export function isFromRenewingMist(event: HealEvent) {
+  return HasRelatedEvent(event, RENEWING_MIST_GOM);
+}
+
+export function isFromVivify(event: HealEvent) {
+  return HasRelatedEvent(event, VIVIFY_GOM);
+}
+
+export function isFromSheilunsGift(event: HealEvent) {
+  return HasRelatedEvent(event, SHEILUNS_GIFT_GOM);
+}
+
+export function isFromRevival(event: HealEvent) {
+  return HasRelatedEvent(event, REVIVAL_GOM);
+}
+
+export function isFromZenPulse(event: HealEvent) {
+  return HasRelatedEvent(event, ZEN_PULSE_GOM);
+}
+
+export function isFromExpelHarm(event: HealEvent) {
+  return HasRelatedEvent(event, EXPEL_HARM_GOM);
+}
+
+export function isFromSoothingMist(event: HealEvent) {
+  return HasRelatedEvent(event, SOOM_GOM);
+}
```

Comment:
Would it make sense to add the `!isFromEssenceFont` check to these functions since they are mutually exclusive?

Reply:
Yeah i'll move them in there